framework/mesh/mesh_generator/distributed_mesh_generator.h

```diff
@@ -0,0 +1,112 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include "framework/mesh/mesh_generator/mesh_generator.h"
+
+namespace opensn
+{
+
+/**
+ * @class opensn::DistributedMeshGenerator
+ * @brief Generates and partitions a mesh on location 0, distributing the partitions via MPI.
+ *
+ * This class is responsible for generating a mesh, partitioning it, and distributing the
+ * individual partitions to different MPI locations. The mesh is generated on location 0,
+ * partitioned into multiple parts, serialized, and distributed to all other MPI ranks.
+ *
+ * The mesh data is serialized into a `ByteArray` for efficient MPI communication and
+ * deserialized at the receiving locations to reconstruct the mesh locally.
+ *
+ * @note This class is derived from the MeshGenerator class.
```

Comment:
1. we don't do `@class` in the doxygen comment, this is inconsistent.
2. Tag format is `\tag`, not `@tag` -> inconsistent.
3. `@note` is cpt. Obvious

Reply:
It would be nice if there was a tool to check Doxygen comment style automatically on PRs, but I don't think such a thing exists. We should probably consider `-Wdocumentation` at some point for `clang`.

Reply:
@wdhawkins Did you do the doco with chatGPT. if so, can you try with the above explicit requirements that David stated? I will have the same issue when I do the entire code with the batch system in a couple of weeks. So, if you find a prompt that works, please email it to me. Thank could save some headache down the road

Reply:
I did use ChatGPT. Let me see if I can make it do it according to our style.